client/me/account/main.jsx

```diff
@@ -699,19 +699,18 @@ class Account extends React.Component {
 		return formName ? this.state.formsSubmitting[ formName ] : this.state.submittingForm;
 	}
 
-	handleSubmitSuccess( response, formName = '' ) {
+	async handleSubmitSuccess( response, formName = '' ) {
 		if ( ! this.hasUnsavedUserSettings( ACCOUNT_FIELDS.concat( INTERFACE_FIELDS ) ) ) {
 			this.props.markSaved();
 		}
 
 		if ( this.state.redirect ) {
-			user()
-				.clear()
-				.then( () => {
-					// Sometimes changes in settings require a url refresh to update the UI.
-					// For example when the user changes the language.
-					window.location = this.state.redirect + '?updated=success';
-				} );
+			await clearStore();
+
+			// Sometimes changes in settings require a url refresh to update the UI.
+			// For example when the user changes the language.
+			window.location = this.state.redirect + '?updated=success';
```

Comment:
Seems we don't really redirect here in a generic sense, but always just reload with a query param. That means there could be a boolean `this.state.reload` and we could:
```js
window.location = '?updated=success';
```
to reload.

Just thinking aloud to share an idea, you don't need to act on it at all 🙂